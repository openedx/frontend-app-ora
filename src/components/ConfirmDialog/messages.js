import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  goBack: {
    defaultMessage: 'Go back',
    description: 'Label for the cancel button',
    id: 'frontend-app-ora.ConfirmDialog.goBack',
  },
});

export default messages;
