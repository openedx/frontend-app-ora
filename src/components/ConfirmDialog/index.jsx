import React from 'react';
import PropTypes from 'prop-types';

import { ActionRow, AlertModal } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import ActionButton from 'components/ActionButton';

import messages from './messages';

const ConfirmDialog = ({
  title,
  description,
  action,
  isOpen,
  close,
}) => (
  <AlertModal
    title={title}
    isOpen={isOpen}
    footerNode={(
      <ActionRow>
        <ActionButton variant="light" onClick={close}>
          <FormattedMessage {...messages.goBack} />
        </ActionButton>
        <ActionButton {...action} />
      </ActionRow>
    )}
  >
    {description}
  </AlertModal>
);
ConfirmDialog.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  action: PropTypes.shape({
    onClick: PropTypes.func.isRequired,
  }).isRequired,
  isOpen: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
};

export default ConfirmDialog;
