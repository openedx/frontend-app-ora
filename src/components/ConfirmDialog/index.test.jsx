import { shallow } from '@edx/react-unit-test-utils';

import ConfirmDialog from './index';

describe('<ConfirmDialog />', () => {
  const props = {
    title: 'Title',
    description: 'Description',
    action: {
      onClick: jest.fn().mockName('onClick'),
    },
    isOpen: true,
    close: jest.fn().mockName('close'),
  };
  it('renders correctly', () => {
    const wrapper = shallow(<ConfirmDialog {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
