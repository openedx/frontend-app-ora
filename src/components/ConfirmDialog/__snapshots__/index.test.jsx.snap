// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmDialog /> renders correctly 1`] = `
<AlertModal
  footerNode={
    <ActionRow>
      <ActionButton
        onClick={[MockFunction close]}
        variant="light"
      >
        <FormattedMessage
          defaultMessage="Go back"
          description="Label for the cancel button"
          id="frontend-app-ora.ConfirmDialog.goBack"
        />
      </ActionButton>
      <ActionButton
        onClick={[MockFunction onClick]}
      />
    </ActionRow>
  }
  isOpen={true}
  title="Title"
>
  Description
</AlertModal>
`;
